hal/armv7a/cpu.c

```diff
@@ -199,3 +200,14 @@ char *hal_cpuFeatures(char *features, unsigned int len)
 
 	return features;
 }
+
+
+void hal_cpuTlsSet(hal_tls_t *tls, cpu_context_t *ctx)
+{
+	/* In theory there should be 8-byte thread control block but
+	 * it's stored elsewhere so we need to subtract 8 from the pointer
+	 */
+	ptr_t ptr = (ptr_t)tls->tls_base - 8;
+	__asm__ volatile("mcr p15, 0, %[value], cr13, cr0, 3;" ::[value] "r"(ptr)
+					 : "r0");
```

Comment:
The `r0` is not needded in the clobbers list. You are not using the register `r0` explicitly.
```suggestion
	__asm__ volatile("mcr p15, 0, %[value], cr13, cr0, 3;" ::[value] "r"(ptr));
```

Reply:
Removed `r0` from clobbers